store/postgres/src/deployment.rs

```diff
@@ -775,7 +775,29 @@ pub fn fail(
 ) -> Result<(), StoreError> {
     let error_id = insert_subgraph_error(conn, error)?;
 
-    update_deployment_status(conn, id, SubgraphHealth::Failed, Some(error_id))?;
+    update_deployment_status(conn, id, SubgraphHealth::Failed, Some(error_id), None)?;
+
+    Ok(())
+}
+
+pub fn update_non_fatal_errors(
+    conn: &PgConnection,
+    deployment_id: &DeploymentHash,
+    health: SubgraphHealth,
+    non_fatal_errors: Option<Vec<SubgraphError>>,
```

Comment:
If you make this an `Option<&[SubgraphError]>` the caller doesn't need to clone - it's not a big deal in this case since errors happen rarely, but on hotter code paths it would make a difference